reconcile/ocm_additional_router.py

```diff
@@ -0,0 +1,105 @@
+import sys
+import logging
+import json
+
+import reconcile.queries as queries
+
+from reconcile.status import ExitCodes
+from reconcile.utils.ocm import OCMMap
+
+QONTRACT_INTEGRATION = 'ocm-additional-router'
+
+
+def fetch_current_state(clusters):
+    settings = queries.get_app_interface_settings()
+    ocm_map = OCMMap(clusters=clusters, integration=QONTRACT_INTEGRATION,
+                     settings=settings)
+
+    current_state = []
+    for cluster in clusters:
+        cluster_name = cluster['name']
+        ocm = ocm_map.get(cluster_name)
+        routers = ocm.get_additional_routers(cluster_name)
+        for router in routers:
+            router['cluster'] = cluster_name
+            current_state.append(router)
+
+    return ocm_map, current_state
+
+
+def fetch_desired_state(clusters):
+    desired_state = []
+    for cluster in clusters:
+        cluster_name = cluster['name']
+        router = cluster['additionalRouter']
+        listening = 'internal' if router['private'] else 'external'
+        item = {
+            'listening': listening,
+            'cluster': cluster_name
+        }
+        selectors = cluster['additionalRouter']['route_selectors']
+        if selectors:
+            item['route_selectors'] = json.loads(selectors)
+        desired_state.append(item)
+
+    return desired_state
+
+
+def calculate_diff(current_state, desired_state):
+    diffs = []
+    err = False
+    for d in desired_state:
+        c = [c for c in current_state
+             if d.items() <= c.items()]
+        if not c:
+            d['action'] = 'create'
+            diffs.append(d)
+
+    for c in current_state:
+        d = [d for d in desired_state
+             if d.items() <= c.items()]
+        if not d:
+            c['action'] = 'delete'
+            diffs.append(c)
+
+    return diffs, err
+
+
+def sort_diffs(diff):
+    """ Sort diffs so we delete first and create later """
+    if diff['action'] == 'delete':
+        return 1
+    else:
+        return 2
+
+
+def act(dry_run, diffs, ocm_map):
+    diffs.sort(key=sort_diffs)
+    for diff in diffs:
```

Comment:
Please, don't use loop variables that differ from the iterated object by just one character. In this context, `for d in diffs` is perfectly understandable.

Reply:
fwiw, this is very common across the code base, and specifically for ocm integrations